src/encoding.rs

```diff
@@ -140,13 +140,23 @@ pub fn deserialize_ergo_tree_constant(c: &Constant) -> Result<P2SAddressString>
     Ok(encoder.address_to_str(&address))
 }
 
+// Takes an Ergo address and encodes it as a mainnet address
+// if it fails then it tries to recover by using the testnet prefix
+pub fn encode_address(address_str: &ErgoAddressString) -> Result<Address> {
```

Comment:
In our code String -> Address transformation is usually referred to as parsing or decoding(rarely). Could you please rename the function for consistency?